docs/how_to.md

```diff
@@ -332,6 +333,66 @@ the header will be the one specified in the terraform configuration ```request h
 
 *Note: Currently, parameters of type 'header' are only supported on an operation level*
 
+###### <a name="xTerraformResourcePollEnabled">x-terraform-resource-poll-enabled</a> 
+
+This extension allows the service provider to enable the polling mechanism in the OpenAPI Terraform provider for asynchronous
+operations. In order for this to work, the following must be met:
+
+- The resource definition must have a read-only field that defines the status of the resource. By default, if a string field caThis can be a field called
+called 'status' is present in the resource schema definition that field will be used to track the different statues of the resource. Alternatively,
+a field can be marked to serve as the status field adding the 'x-terraform-field-status'. This field will be used as the status
+field even if there is another field named 'status'. This gives service providers flexibility to name their status field the
+way they desire. More details about the 'x-terraform-field-status' extension can be found in the [Attribute details](#attributeDetails) section.
+- The polling mechanism required two more extensions to work which define the expected 'status' values for both target and 
+pending statuses. These are:
+
+  - **x-terraform-resource-poll-target-statuses**: (type: string) Comma separated values - Defines the statuses on which the resource state will be considered 'completed'
+  - **x-terraform-resource-poll-pending-statuses**: (type: string) Comma separated values - Defines the statuses on which the resource state will be considered 'in progress'.
+Any other state returned that returned but is not part of this list will be considered as a failure and the polling mechanism
+will stop its execution accordingly.
+
+In the example below, the response with HTTP status code 202 has the extension defined with value 'true' meaning 
+that the OpenAPI Terraform provider will treat this response as asynchronous. Therefore, the provider will perform
+continues calls to the resource's instance GET operation and will use the value from the resource 'status' property to
+determine the state of the resource:
+
+````
+  /v1/lbs:
+    post:
+      ...
+      responses:
+        202: # Accepted
+          x-terraform-resource-poll-enabled: true # [type (bool)] - this flags the response as trully async. Some resources might be async too but may require manual intervention from operators to complete the creation workflow. This flag will be used by the OpenAPI Service provider to detect whether the polling mechanism should be used or not. The flags below will only be applicable if this one is present with value 'true'
```

Comment:
Question: I know we're going for backwards compatibility, but I wonder if this should default to true and need to be switched off - it would also switch off if none of the supporting extensions are present, which would handle the backwards-compatibility.

Reply:
As agreed, leaving current configuration so we let the user know what the issue is (in case they are missing some required extensions) instead of falling back to a synchronous behaviour and not being able to let the user know that the swagger config is wrong at runtime.